python/ray/serve/_private/deployment_state.py

```diff
@@ -2284,6 +2287,16 @@ def shutdown(self):
         # TODO(jiaodong): Need to add some logic to prevent new replicas
         # from being created once shutdown signal is sent.
 
+    def is_shutdown(self) -> bool:
```

Comment:
All the `is_shutdown` methods also return `True` if no Serve application has started yet. Could we rename the functions:

```suggestion
    def is_ready_for_shutdown(self) -> bool:
```